src/presign/round_one.rs

```diff
@@ -22,16 +22,39 @@ use serde::{Deserialize, Serialize};
 use std::fmt::Debug;
 use zeroize::ZeroizeOnDrop;
 
+use super::round_three::SecretBigNumber;
+
+#[derive(Clone, ZeroizeOnDrop)]
+pub(crate) struct SecretNonce(Nonce);
+
+impl SecretNonce {
+    pub fn get_nonce(&self) -> &Nonce {
+        &self.0
+    }
+}
+
+impl From<Nonce> for SecretNonce {
```

Comment:
I think we should use a constructor instead of this From instance. This will make our conversions more explicit.